src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -113,27 +114,78 @@ def createStageOutCommand(self, sourcePFN, targetPFN, options=None, checksums=No
                 copyCommandDict['checksum'] = "-K adler32"
 
         copyCommandDict['options'] = ' '.join(unknown)
-
         copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
         copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
 
         copyCommand = self.copyCommand % copyCommandDict
         result += copyCommand
 
+        logging.debug("Actual command which failed: %s", copyCommand)
+
         if _CheckExitCodeOption:
             result += """
             EXIT_STATUS=$?
             echo "gfal-copy exit status: $EXIT_STATUS"
             if [[ $EXIT_STATUS != 0 ]]; then
-               echo "ERROR: gfal-copy exited with $EXIT_STATUS"
-               echo "Cleaning up failed file:"
-               %s
+                echo "ERROR: gfal-copy exited with $EXIT_STATUS"
+                echo "Source PFN: {source}"
+                echo "Target PFN: {destination}"
+                echo "Cleaning up failed file: {remove_command}"
```

Comment:
This command is now missing the file removal logic, as you deleted that in favor of simply printing it.

Reply:
You might have missed this comment.